src/components/Chart/ReferenceAreaTick.tsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react'
+import { Rectangle } from 'recharts'
+
+import { useTheme } from '../../styles'
+import { splitIntoLines } from '../../util/string'
+import { ReferenceAreaWithPercents, TickProps } from './model'
+
+const MAX_CHARS_PER_LINE = 8
+
+export interface ReferenceTickProps extends TickProps {
+  refTicks: Map<number, ReferenceAreaWithPercents<any>>
+}
+
+export function ReferenceAreaTick(props: ReferenceTickProps) {
+  const { x, y, payload, refTicks, height } = props
+
+  const ref = refTicks.get(+payload.value)
+  const margin = 5
+
+  const theme = useTheme()
+
+  const nameLines = splitIntoLines(ref.name, MAX_CHARS_PER_LINE)
+
+  return (
+    <>
+      <text
+        x={x}
+        y={y}
+        dx={15}
+        dy={15}
+        textAnchor='start'
+        fill={ref.tickColor ?? ref.color}
+        style={{ fontWeight: 'bold' }}
+      >
+        {nameLines.map(
+          (namePart, i) =>
+            namePart && (
+              <tspan key={namePart} dx={15} dy={(i + 1) * 15} x={x} y={y}>
```

Comment:
:mage_woman: (_magic numbers_)
I think it's good to give a variable to this `15` because it is used elsewhere and give it some meaning